core/git_mixins/checkout_discard.py

```diff
@@ -23,8 +23,11 @@ def checkout_file(self, fpath):
         """
         self.git("checkout", "--", fpath)
 
-    def checkout_ref(self, ref):
+    def checkout_ref(self, ref, fpath=None):
```

Comment:
Maybe we should decorate this with the `destructive` decorator? E.g:

    @util.actions.destructive(description="discard uncommitted changes to file")
    def checkout_ref(self, ref, fpath=None):
        ...

And, we could also do it with the above couple of functions...

Reply:
It makes sense.